Samples/NetworkMediaStreamer/CameraRTPStreamerApp/CameraRTPStreamerApp.cpp

```diff
@@ -0,0 +1,430 @@
+    si.dwYCountChars = 10;
+    si.dwFlags = STARTF_USESHOWWINDOW | STARTF_USEPOSITION | STARTF_PREVENTPINNING | STARTF_USECOUNTCHARS;
+    si.wShowWindow = SW_SHOWNOACTIVATE;
+    // Start the child process. 
+    if (!CreateProcess(NULL,   // No module name (use command line)
+        (LPWSTR)cmdline.c_str(),        // Command line
+        NULL,           // Process handle not inheritable
+        NULL,           // Thread handle not inheritable
+        FALSE,          // Set handle inheritance to FALSE
+        CREATE_NEW_CONSOLE,              // No creation flags
+        NULL,           // Use parent's environment block
+        NULL,           // Use parent's starting directory 
+        &si,            // Pointer to STARTUPINFO structure
+        &pi)            // Pointer to PROCESS_INFORMATION structure
+        )
+    {
+        check_win32(GetLastError());
+    }
+    return pi;
+
+}
+
+void StopLoggerConsole(PROCESS_INFORMATION pi)
+{
+    TerminateProcess(pi.hProcess, 0);
+    // Close process and thread handles. 
+    CloseHandle(pi.hProcess);
+    CloseHandle(pi.hThread);
+}
+
+winrt::hstring GetDeviceSelectionFromUser()
+{
+    auto filteredDevices = DeviceInformation::FindAllAsync(DeviceClass::VideoCapture).get();
+    if (!filteredDevices.Size())
+    {
+        throw_hresult(MF_E_NO_CAPTURE_DEVICES_AVAILABLE);
+    }
+    int selection = 1;
+    auto deviceIter = filteredDevices.First();
+    while (deviceIter.HasCurrent())
+    {
+        auto device = deviceIter.Current();
+        std::wcout << std::endl << selection++ << L". " << device.Name().c_str() << L":" << device.Id().c_str();
+        deviceIter.MoveNext();
+    }
+    std::cout << "\nEnter Selection:";
+    std::cin >> selection; selection--;
+    return filteredDevices.GetAt(selection).Id();
+}
+
+void GetUserSelectionAndSetMediaFormat(MediaCapture& mc)
+{
+    auto formats = mc.VideoDeviceController().GetAvailableMediaStreamProperties(MediaStreamType::VideoRecord);
+    IMediaEncodingProperties selectedProp(nullptr);
+    uint32_t idx = 1;
+    for (auto f : formats)
+    {
+        auto format = f.try_as<VideoEncodingProperties>();
+        float fr = (float)format.FrameRate().Numerator() / (float)format.FrameRate().Denominator();
+        std::wcout << L"\n" << idx++ << L". " << format.Width() << L"x" << format.Height() << L"@" << /*std::setprecision(2) <<*/ fr << L":" << format.Subtype().c_str();
+    }
+    std::cout << "Enter format choice: ";
+    std::cin >> idx; idx--;
+    selectedProp = formats.GetAt(idx);
+    mc.VideoDeviceController().SetMediaStreamPropertiesAsync(MediaStreamType::VideoRecord, selectedProp).get();
+}
+
+int main()
+{
+    std::cout << "VideoSteamer App \n";
+    PROCESS_INFORMATION loggerConsole;
+    try
+    {
+
+        std::wofstream fileLogger("RTSPServer.log");
+        fileLogger << L"StartLogging..\n";
+        loggerConsole = StartLoggerConsole(L"RTSPServer.log");
+
+        auto mc = MediaCapture();

```

Comment:
recommend register callback to FailedEvent, pipeline returns catastrophic failure here.